### ISSUE TITLE
Problems with PGFPlotsX during the install

### ISSUE THREAD

**Issue by ll7** [JuliaReinforcementLearning/ReinforcementLearning.jl#301]
I tried to install ReinforcementLearning.jl with `add ReinforcementLearning`. The following error occured:

```julia
Building PGFPlotsX → `~/.julia/scratchspaces/44cfe95a-1eb2-52ea-b672-e2afdf69b78f/e8961c48743b383f01ba13187b9cfcfc080fc004/build.log`
ERROR: Error building `PGFPlotsX`: 
ERROR: LoadError: UndefVarError: info not defined
Stacktrace:
 [1] top-level scope
   @ ~/.julia/packages/PGFPlotsX/V7bWm/deps/build.jl:5
 [2] include(fname::String)
   @ Base.MainInclude ./client.jl:444
 [3] top-level scope
   @ none:5
in expression starting at /home/luttkule/.julia/packages/PGFPlotsX/V7bWm/deps/build.jl:4
```

to get further information, I did the following:

```julia
julia> import Pkg; Pkg.precompile()
Precompiling project...
  ✗ PGFPlotsX
  ✗ ReinforcementLearning
  0 dependencies successfully precompiled in 12 seconds (190 already precompiled)

ERROR: The following 1 direct dependency failed to precompile:

ReinforcementLearning [158674fc-8238-5cab-b5ba-03dfc80d1318]

┌ Warning: Package ReinforcementLearning does not have Dates in its dependencies:
│ - If you have ReinforcementLearning checked out for development and have
│   added Dates as a dependency but haven't updated your primary
│   environment's manifest file, try `Pkg.resolve()`.
│ - Otherwise you may need to report an issue with ReinforcementLearning
└ Loading Dates into ReinforcementLearning from project dependency, future warnings for ReinforcementLearning are suppressed.
WARNING: could not import ReinforcementLearningBase.interact! into ReinforcementLearning
WARNING: could not import ReinforcementLearningBase.getstate into ReinforcementLearning
WARNING: could not import ReinforcementLearningBase.plotenv into ReinforcementLearning
WARNING: could not import ReinforcementLearningBase.actionspace into ReinforcementLearning
WARNING: could not import ReinforcementLearningBase.sample into ReinforcementLearning
ERROR: LoadError: LoadError: LoadError: UndefVarError: functorm not defined
Stacktrace:
  [1] var"@treelike"(__source__::LineNumberNode, __module__::Module, args::Vararg{Any, N} where N)
    @ Flux ~/.julia/packages/Flux/sY3yx/src/functor.jl:58
  [2] include(mod::Module, _path::String)
    @ Base ./Base.jl:386
  [3] include(x::String)
    @ ReinforcementLearning ~/.julia/packages/ReinforcementLearning/qSdCS/src/ReinforcementLearning.jl:1
  [4] top-level scope
    @ ~/.julia/packages/ReinforcementLearning/qSdCS/src/ReinforcementLearning.jl:31
  [5] include
    @ ./Base.jl:386 [inlined]
  [6] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
    @ Base ./loading.jl:1213
  [7] top-level scope
    @ none:1
  [8] eval
    @ ./boot.jl:360 [inlined]
  [9] eval(x::Expr)
    @ Base.MainInclude ./client.jl:446
 [10] top-level scope
    @ none:1
in expression starting at /home/luttkule/.julia/packages/ReinforcementLearning/qSdCS/src/flux.jl:12
in expression starting at /home/luttkule/.julia/packages/ReinforcementLearning/qSdCS/src/flux.jl:12
in expression starting at /home/luttkule/.julia/packages/ReinforcementLearning/qSdCS/src/ReinforcementLearning.jl:1
Stacktrace:
 [1] pkgerror(msg::String)
   @ Pkg.Types /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/Types.jl:55
 [2] precompile(ctx::Pkg.Types.Context; internal_call::Bool, strict::Bool, kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Pkg.API /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:1244
 [3] precompile
   @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:920 [inlined]
 [4] #precompile#196
   @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:918 [inlined]
 [5] precompile()
   @ Pkg.API /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:918
 [6] top-level scope
   @ REPL[4]:1
```

I also tried:

```julia
julia> import Pkg; Pkg.resolve()
  No Changes to `~/git/SCPS-LL/scpsLL/Project.toml`
  No Changes to `~/git/SCPS-LL/scpsLL/Manifest.toml`
```

I am using ubuntu 20.04, Julia 1.6.1.

I don't know how to fix any of the issues.

**Comment by ll7** [comment]
I opened an issue at https://github.com/KristofferC/PGFPlotsX.jl/issues/268 because it might be more related to the problems of PGFPlotsX

**Comment by ll7** [comment]
https://github.com/KristofferC/PGFPlotsX.jl/issues/268#issuecomment-845864076

**Comment by ll7** [comment]
I am still having trouble to use AutomotiveSimulator.jl and ReinforcementLearning.jl simultaneously. It appears to bre related to requirement of ReinforcementLearning.jl, such that it needs an old version of `  [8314cec4] ↓ PGFPlotsX v1.2.10 ⇒ v0.2.2`.

**Comment by ll7** [comment]
Thank you. This sounds great. Would it make more sense to ask `AutomotiveSimulator.jl` to update to v0.24?

**Comment by findmyway** [comment]
I just tried to add support for `Distributions@v0.23` it seems there will be another unsatisfiable requirements for `SpecialFunctions`.

But with `AutomotiveSimulator#master` (which contains compat of `Distributions@v0.24`), it seems to be ok.

```
ERROR: Unsatisfiable requirements detected for package SpecialFunctions [276daf66]:
 SpecialFunctions [276daf66] log:
 ├─possible versions are: 0.7.0-1.4.2 or uninstalled
 ├─restricted by compatibility requirements with CUDA [052768ef] to versions: 1.3.0-1.4.2
 │ └─CUDA [052768ef] log:
 │   ├─possible versions are: 0.1.0-3.2.1 or uninstalled
 │   └─restricted to versions 3 by ReinforcementLearningZoo [d607f57d], leaving only versions 3.0.0-3.2.1
 │     └─ReinforcementLearningZoo [d607f57d] log:
 │       ├─possible versions are: 0.4.1 or uninstalled
 │       └─ReinforcementLearningZoo [d607f57d] is fixed to version 0.4.1
 └─restricted by compatibility requirements with Distributions [31c24e10] to versions: 0.8.0-0.10.3 — no versions left
   └─Distributions [31c24e10] log:
     ├─possible versions are: 0.16.0-0.25.1 or uninstalled
     ├─restricted to versions 0.23-0.25 by ReinforcementLearningZoo [d607f57d], leaving only versions 0.23.0-0.25.1
     │ └─ReinforcementLearningZoo [d607f57d] log: see above
     ├─restricted by compatibility requirements with StatsBase [2913bbd2] to versions: 0.23.2-0.25.1 or uninstalled, leaving only versions: 0.23.2-0.25.1
     │ └─StatsBase [2913bbd2] log:
     │   ├─possible versions are: 0.24.0-0.33.8 or uninstalled
     │   ├─restricted to versions 0.32-0.33 by ReinforcementLearningEnvironments [25e41dd2], leaving only versions 0.32.0-0.33.8
     │   │ └─ReinforcementLearningEnvironments [25e41dd2] log:
     │   │   ├─possible versions are: 0.6.1 or uninstalled
     │   │   └─ReinforcementLearningEnvironments [25e41dd2] is fixed to version 0.6.1
     │   └─restricted by compatibility requirements with DataStructures [864edb3b] to versions: 0.33.1-0.33.8 or uninstalled, leaving only versions: 0.33.1-0.33.8
     │     └─DataStructures [864edb3b] log:
     │       ├─possible versions are: 0.9.0-0.18.9 or uninstalled
     │       └─restricted to versions 0.18 by ReinforcementLearningZoo [d607f57d], leaving only versions 0.18.0-0.18.9
     │         └─ReinforcementLearningZoo [d607f57d] log: see above
     └─restricted by compatibility requirements with AutomotiveSimulator [6aa42d20] to versions: 0.23.0-0.23.12, leaving only versions: 0.23.2-0.23.12
       └─AutomotiveSimulator [6aa42d20] log:
         ├─possible versions are: 0.1.0-0.1.1 or uninstalled
         └─restricted to versions * by an explicit requirement, leaving only versions 0.1.0-0.1.1
```

> I thought that the problem might be related to PGFPlotsX but that is not the case?

Yes, there's nothing related to PGFPlotsX here. Installing the latest AutomotiveSimulator.jl will force a downgrade of ReinforcementLearning.jl, which relies on an old version of PGFPlotsX.

In summary:

1. AutomotiveSimulator needs to tag a new version.
2. RLZoo.jl needs to tag a new version which includes the compat of `Distributions@v0.24`
